### PR TITLE
Document invariants

### LINKED ISSUE #6: Mention invariants
...and their interaction with greasing. Invariants are the properties of the protocol that you don't want to protect.

### PR DESCRIPTION
**Pull request by martinthomson**
I also adjusted some of the language around greasing based on Mirja's
feedback.

Closes #6.

/cc @britram, @mirjak

### CODE REVIEW

File: draft-thomson-use-it-or-lose-it.md [hunk 1]
```diff
@@ -379,13 +377,32 @@ the mechanisms it safeguards.  Though it has been effective at revealing
...
+Documenting aspects of the protocol that cannot or will not change as
+extensions or new versions are added can be a useful exercise. Understanding
+what aspects of a protocol are invariant can help guide the process of
+protecting those parts of the protocol that might change.
+
+Like greasing, a declaration of protocol invariants is useful only to the
+extent that implementations of the protocol are willing to admit the
+possibility of new uses for the protocol.  A protocol that declares protocol
+invariants relies on implementations understanding and respecting those
+invariants.
```
Comment (mirjak) [inline draft-thomson-use-it-or-lose-it.md]: Not sure I really understand what you want to say here. When you talk about implementations, is that endpoint implementations or any implementation that may interact with the protocol? Also how could an implementation not admit an invariant? For me documenting invariants was not really intended to guide implementation (so far) but rather have an explicit step in the design of a protocol.

**Comment by mirjak** [comment]
Also not sure if putting this as a subsection in 4 is the best place to put it. However, as I indicated in the discussion in the StackEvo meeting, putting it somewhere else would probably need a bit of restructuring of the whole document...

File: draft-thomson-use-it-or-lose-it.md (hunk 1 above)
Comment (martinthomson) [reply draft-thomson-use-it-or-lose-it.md]: I took another spin at this.  Let me know if this is better.

**Comment by martinthomson** [comment]
I appreciate the point about structure here, but it's not particularly easy to throw everything up in the air and start over.  If you have some ideas about how to move toward a better structure, I can try to shuffle things around a little.

**Pull request merged**